### DISCUSSION TITLE
Brew install from external http url is not working

### DISCUSSION THREAD

**Discussion by synacker** [Homebrew/discussions#1190] [Everyday usage]
I have an issue for my project - https://github.com/synacker/daggy/issues/24 
The problem, in short description, in installing formula from external url by next command:
````
brew install https://raw.githubusercontent.com/synacker/daggy_homebrew/master/daggy.rb
````
In my environment this command works fine, but for several users of my application don't work - there is next error:
````
Calling Non-checksummed download of daggy formula file from an arbitrary URL is disabled! Use 'brew extract' or 'brew create' and 'brew tap-new' to create a formula file in a tap on GitHub instea
````
What is the problem can be?

**Comment by carlocab** [comment]
The problem is this:
```
Calling Non-checksummed download of daggy formula file from an arbitrary URL is disabled!
```
which means that
```bash
brew install https://raw.githubusercontent.com/synacker/daggy_homebrew/master/daggy.rb
```
should not work. I don't know why it works in your environment. It's likely that you're using an outdated version of `brew`. You can check whether it is with `brew --version`.

In order to use your formula with the latest version of `brew`, I think the following should work:
```bash
brew tap synacker/daggy https://github.com/synacker/daggy_homebrew.git
brew install synacker/daggy/daggy
```

**Reply by synacker** [reply]
@carlocab  Thank you for answer, I will try tap formula!

**Comment by SMillerDev** [comment]
> brew install from a URL has been deprecated to improve the security of brew install.

https://brew.sh/2020/05/29/homebrew-2.3.0/

---

That feature was deprecated half a year ago and has been removed since, anyone running a modern/supported version of brew will not be able to do this.

**Reply by synacker** [reply]
@SMillerDev thank you for reply!
But, what is alternative way for installing formula that still not in the main repo?

**Reply by carlocab** [reply]
See https://github.com/Homebrew/discussions/discussions/1190#discussioncomment-565328. Alternatively:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/
